File: Finance/Models/FinancialGoal.cs
using System.ComponentModel.DataAnnotations;

namespace Finance.Models
{
    public class FinancialGoal
    {
        [Key]
        public int Id { get; set; } // Chave primária
        public decimal Amount { get; set; } // Valor da meta financeira
        public string Name { get; set; } // Nome da meta
        public DateTime CreatedOn { get; set; } // Data de criação da meta
        public DateTime? EstimatedCompletion { get; set; } // Previsão de término (opcional)
        public string Status { get; set; } // Status da meta (ex: Active, Completed, Cancelled)
        public int Parcelas { get; set; }
        public int ParcelaAtual { get; set; }
        public bool Check { get; set; }
        public string Description { get; set; }
        public decimal InstallmentValue { get; set; }
        public bool IsPaidThisMonth { get; set; }
        public DateTime? LastPaymentDate { get; set; }
        public FinancialGoal()
        {
            // Inicialização de valores padrão
            Amount = 0;
            Name = "Nova Meta Financeira";
            CreatedOn = DateTime.Now;
            EstimatedCompletion = null;
            Status = "Active";
        }
    }
}

File: Finance/ClientApp/src/app/ts/FinancialGoal.ts
export  class FinancialGoal {
    id: number; // Chave primária
    amount: number; // Valor da meta financeira
    name: string; // Nome da meta
    createdOn: Date; // Data de criação da meta
    estimatedCompletion?: Date; // Previsão de término (opcional)
    status: string; // Status da meta (ex: Active, Completed, Cancelled)
    parcelas: string; // Representação das parcelas, presumivelmente como uma string
    parcelaAtual: string; // Indicador da parcela atual
    check: boolean; // Uma marcação de checkbox, por exemplo
    description: string; // Descrição da meta financeira

    constructor(
        id: number,
        amount: number,
        name: string,
        createdOn: Date,
        status: string,
        parcelas: string,
        parcelaAtual: string,
        check: boolean,
        description: string,
        estimatedCompletion?: Date
    ) {
        this.id = id;
        this.amount = amount;
        this.name = name;
        this.createdOn = createdOn;
        this.estimatedCompletion = estimatedCompletion;
        this.status = status;
        this.parcelas = parcelas;
        this.parcelaAtual = parcelaAtual;
        this.check = check;
        this.description = description;
    }
}